src/Infrastructure.VS/LanguageDetection/SonarLanguageRecognizer.cs

```diff
@@ -112,5 +119,37 @@ private static bool IsTypeScriptDocument(IEnumerable<IContentType> contentTypes)
         {
             return contentTypes.Any(type => type.IsOfType("TypeScript"));
         }
+
+        public AnalysisLanguage? GetAnalysisLanguageFromExtension(string fileName)
+        {
+            var extension = GetNormalizedExtention(fileName);
+
+            if (JavascriptSupportedExtensions.Contains(extension)) { return AnalysisLanguage.Javascript; }
+            if (IsContentType(TypeScriptTypeName, extension)) { return AnalysisLanguage.TypeScript; }
```

Comment:
What happens if a user opens a folder with a .ts file in it? I don't think it will have a content type.

Reply:
We do not check the file's content type but if the file extension is associated with the given type.